magicctapipe/scripts/lst1_magic/lst1_magic_cloud_correction.py

```diff
@@ -0,0 +1,715 @@
+        type=str,
+        required=True,
+        help="Path to an input .h5 DL1 data file",
+    )
+
+    parser.add_argument(
+        "--output_dir",
+        "-o",
+        dest="output_dir",
+        type=str,
+        default="./data",
+        help="Path to a directory where to save an output corrected DL1 file",
+    )
+
+    parser.add_argument(
+        "--config_file",
+        "-c",
+        dest="config_file",
+        type=str,
+        default="./resources/config.yaml",
+        help="Path to a configuration file",
+    )
+
+    args = parser.parse_args()
+
+    subarray_info = SubarrayDescription.from_hdf(args.input_file)
+
+    tel_descriptions = subarray_info.tel
+    camgeom = {}
+    for telid, telescope in tel_descriptions.items():
+        camgeom[telid] = telescope.camera.geometry
+
+    optics_table = read_table(
+        args.input_file, "/configuration/instrument/telescope/optics"
+    )
+    focal_eff = {}
+
+    for telid, telescope in tel_descriptions.items():
+        optics_row = optics_table[optics_table["optics_name"] == telescope.name]
+        if len(optics_row) > 0:
+            focal_eff[telid] = optics_row["effective_focal_length"][0] * u.m
+        else:
+            raise ValueError(f"No optics data found for telescope: {telescope.name}")
+
+    with open(args.config_file, "r") as file:
+        config = yaml.safe_load(file)
+
+    tel_ids = config["mc_tel_ids"]
+
+    correction_params = config.get("cloud_correction", {})
+    max_gap_lidar_shots = u.Quantity(
+        correction_params.get("max_gap_lidar_shots")
+    )  # set it to 900 seconds
+    lidar_report_file = correction_params.get(
+        "lidar_report_file"
+    )  # path to the lidar report
+    nlayers = correction_params.get("number_of_layers")
+    cmf = correction_params.get("cleaning_multiplication_factor")
+
+    dl1_params = read_table(args.input_file, "/events/parameters")
+
+    mean_subrun_timestamp = np.mean(dl1_params["timestamp"])
+    mean_subrun_zenith = np.mean(90.0 - np.rad2deg(dl1_params["pointing_alt"]))
+
+    cloud_params = lidar_cloud_interpolation(
+        mean_subrun_timestamp, max_gap_lidar_shots, lidar_report_file
+    )
+
+    Hc = u.Quantity(cloud_params[0], u.m)
+    dHc = u.Quantity(cloud_params[1] - cloud_params[0], u.m)
+    vertical_trans = u.Quantity(cloud_params[2])
+    trans = vertical_trans ** (1 / np.cos(np.deg2rad(mean_subrun_zenith)))
+
+    dfs = []
+
+    for tel_name, tel_id in tel_ids.items():
+        if tel_id != 0:  # Only process telescopes that have a non-zero ID
+            # Read images for each telescope
+            image_node_path = "/events/dl1/image_" + str(tel_id)
+            try:
+                dl1_images = read_table(args.input_file, image_node_path)
+            except tables.NoSuchNodeError:
+                logger.info(f"\nNo image for telescope with ID {tel_id}. Skipping.")
+                dl1_images = None
+
+            df = process_telescope_data(
```

Comment:
you pass the config to the function, so the parameters like tel_ids, nlayers, cmf should be recovered from the config inside this function instead of being passed additionally.